libraries/provider_systemd_service.rb

```diff
@@ -0,0 +1,19 @@
+
+require_relative 'provider_systemd_unit'
+require_relative 'resource_systemd_service'
+
+class Chef::Provider
+  class SystemdService < Chef::Provider::SystemdUnit
+    # rubocop: disable AbcSize
+    def load_current_resource
```

Comment:
Ideally `load_current_resource` would set current_resource to the existing state of the unit file.